backend/webapi/Features/Parties/Demographics.cs

```diff
@@ -89,9 +89,9 @@ public async Task HandleAsync(Command command)
                 party.DomainEvents.Add(new PartyEmailUpdated(party.Id, party.PrimaryUserId, command.Email!));
             }
 
-            party.PreferredFirstName = command.PreferredFirstName;
-            party.PreferredMiddleName = command.PreferredMiddleName;
-            party.PreferredLastName = command.PreferredLastName;
+            party.PreferredFirstName = string.IsNullOrWhiteSpace(command.PreferredFirstName) ? null : command.PreferredFirstName;
+            party.PreferredMiddleName = string.IsNullOrWhiteSpace(command.PreferredMiddleName) ? null : command.PreferredMiddleName;
+            party.PreferredLastName = string.IsNullOrWhiteSpace(command.PreferredLastName) ? null : command.PreferredLastName;
```

Comment:
I'm 100% in support of this (assuming the business doesn't want white-space preferred names), but this isn't quite enough on its own. We also need to update the frontend validators to disallow white-space names. Both because it is better for the UX but also because you now have the possibility that a user enters a valid first name/last name but a white-space last/first name. In that case they would have one name saved to the database and another that gets converted to null. When that user returns to the demographics screen they either get an invalid form that they can't submit even if they didn't change anything (because their last name is null); or the toggle doesn't open and deletes the user's preferred names when they submit (because first name is null)